src/subdomains/supporting/bank/bank-account/bank-account.service.ts

```diff
@@ -56,22 +56,15 @@ export class BankAccountService {
       where: { iban: dto.iban, userData: { id: userDataId } },
       relations: ['userData'],
     });
-    if (existing) {
-      if (existing.active) throw new ConflictException('BankAccount already exists');
 
-      if (!existing.active) {
-        // reactivate deleted bank account
-        existing.active = true;
-        await this.bankAccountRepo.save(existing);
-      }
-
-      return existing;
+    if (existing && !existing.active) {
+      // reactivate deleted bank account
+      existing.active = true;
     }
 
-    const bankAccount = await this.getOrCreateBankAccountInternal(dto.iban, userDataId, kycType);
+    const bankAccount = existing ? existing : await this.getOrCreateBankAccountInternal(dto.iban, userDataId, kycType);
 
-    const update = await this.updateEntity(dto, bankAccount);
-    return this.bankAccountRepo.save(update);
+    return this.updateEntity(dto, bankAccount);
```

Comment:
Maybe just like this?
```ts
const bankAccount = existing ?? await this.getOrCreateBankAccountInternal(dto.iban, userDataId, kycType);
return this.updateEntity(dto, bankAccount);
```
and set `active = true` as default in `CreateBankAccountDto`?